tests/providers/sftp/triggers/test_sftp.py

```diff
@@ -175,13 +175,10 @@ async def test_sftp_trigger_run_trigger_failure_state(self, mock_get_files_by_pa
 
         trigger = SFTPTrigger(path="test/path/", sftp_conn_id="sftp_default", file_pattern="my_test_file")
 
-        expected_event = {"status": "failure", "message": "An unexpected exception"}
-
-        with pytest.raises(Exception):
-            generator = trigger.run()
-            actual_event = await generator.asend(None)
+        generator = trigger.run()
+        actual_event = await generator.asend(None)
```

Comment:
This change seems wrong?

Reply:
I guess this test "worked" because `pytest.raises(Exception)` catch assertion error from `assert TriggerEvent(expected_event) == actual_event` 🙄

Reply:
Merged the solution for this from #38525